### PR TITLE
Lesson 2

### PR DESCRIPTION
**Pull request by Charlie-England**

### CODE REVIEW

**Comment by Charlie-England** [comment]
conflicting files due to my code for grid_printer.py being used in class. I changed grid_printer to make it more readable and removed unnecessary code.

**Comment by PythonCHB** [comment]
to resolve this, you should be able to do:

git pull upstream master.

you will then have conflicts on your laptop.

Open the file, and you'll find both version of the conflicting code. remove the stuff you do'nt want, and commit again.

you could also probably do it in the web editor from here.

If this all goes to heck, we and fix it for you -- or fix it in class.

File: students/charlie_england/session02/fizz_buzz.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Write a program that prints the numbers from 1 to 100 inclusive.
+# But for multiples of three print “Fizz” instead of the number.
+# For the multiples of five print “Buzz” instead of the number.
+# For numbers which are multiples of both three and five print “FizzBuzz” instead.
+
```
Comment (PythonCHB) [inline students/charlie_england/session02/fizz_buzz.py:5]: good, solid solution.

File: students/charlie_england/session02/grid_printer.py [hunk 2]
```diff
@@ -1,29 +1,13 @@
...
-            for _ in range(3):
-                for nc in range(num_columns*2+1):
-                    if nc % 2 == 0:
-                        print("|", end="")
-                    else:
-                        print(" "*9, end="")
-                    if nc == num_columns*2:
-                        print("")
 
+def pri_grid(size = 2, cols = 2, rows = 2): #instead of just 2 parameters, allows you to change the # of rows and cols independently
```
Comment (PythonCHB) [inline students/charlie_england/session02/grid_printer.py:2]: very nice!

File: students/charlie_england/session02/series.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#tests the  sum_series to 7 degrees given a starting parameter of 2 and 1 (default is the fibonnaci series and this changes
+# to the lucas series given the starting parameters of 2 and 1)
+assert sum_series(0,2,1) == 2
+assert sum_series(1,2,1) == 1
+assert sum_series(2,2,1) == 3
+assert sum_series(3,2,1) == 4
+assert sum_series(4,2,1) == 7
+assert sum_series(5,2,1) == 11
+assert sum_series(6,2,1) == 18
+assert sum_series(7,2,1) == 29
```
Comment (PythonCHB) [inline students/charlie_england/session02/series.py:77]: nicely done!

**Review by PythonCHB — APPROVED** [review]
this all looks great -- nice work.

File: students/charlie_england/session02/grid_printer.py [hunk 4]
```diff
@@ -10,4 +10,14 @@ def pri_grid(size = 2, cols = 2, rows = 2): #instead of just 2 parameters, allow
             print((" |" + "  "*size)*cols+" |")
     print((" +" + " -"*size)*cols+" +") 
 
-pri_grid(5,5,10)
\ No newline at end of file
+pri_grid(5,5,10)
+=======
```
Comment (PythonCHB) [inline students/charlie_england/session02/grid_printer.py:14]: this is an artifact of the merge conflict -- you'll get a syntax error if you try to run the code.

As a rule, you want to test your code after "resolving the conflict", before committing the change.

Comment (PythonCHB) [reply students/charlie_england/session02/grid_printer.py:14]: I'm going to merge, so we can move on.

Whether you want to fix this now is up to you -- you've gotten credit for the assignment!

**Comment by PythonCHB** [comment]
Congrats on resolving a merge conflict!

File: students/charlie_england/session02/grid_printer.py (hunk 4 above)
Comment (Charlie-England) [reply students/charlie_england/session02/grid_printer.py:14]: Thanks! still getting used to github and I edited it inside of github. Next time I'll do the git pull upstream master and edit it inside my IDE and make sure to test it. Appreciate the help!

**Pull request merged**